### ISSUE TITLE
Regression in iterating over empty vectors with non-void pointer types

### ISSUE THREAD

**Issue by guitargeek** [wlav/cppyy#281]
Reproducer:
```python
import cppyy

for x in cppyy.gbl.vector["std::string*"]():
    pass
```

This regression caused the failures in the ROOT synchronization PR:
  * https://github.com/root-project/root/pull/17284

With `void*` or any non-pointer types, it still works just fine as far as I can tell!

**Comment by wlav** [comment]
This is fixed with https://github.com/wlav/CPyCppyy/pull/45, right? I added a test and a note in the change log.

**Comment by guitargeek** [comment]
Yes, this is fixed by the PR you linked. Thanks for adding the test!
